app/views/handlers/individual_response.py

```diff
@@ -137,22 +140,29 @@ def handle_get(self):
             self.individual_section_id
         )
 
-        if self._list_item_id:
+        if self._return_to_hub:
+            previous_location_url = url_for("questionnaire.get_questionnaire")
+        else:
             previous_location_url = url_for(
                 "questionnaire.block",
                 list_name=self._list_name,
                 list_item_id=self._list_item_id,
                 block_id=individual_section_first_block_id,
             )
-        else:
-            previous_location_url = url_for("questionnaire.get_questionnaire")
 
         return render_template(
             template="individual_response/interstitial",
             language=self._language,
             previous_location_url=previous_location_url,
         )
 
+    def handle_post(self):
+        if self._list_item_id:
+            return redirect(
+                url_for(".get_individual_response_how", list_item_id=self._list_item_id)
+            )
+        return redirect(url_for(".get_individual_response_who", return_to="hub"))
```

Comment:
At the moment, when there is only 1 non-primary person, this is redirecting to the who route which then redirects to how route, but we should be avoiding that extra redirect as we can work out at this point whether to go to the who route or the how route based on the count of non-primary list items.